agent/handlers/credentials/handler.go

```diff
@@ -81,9 +96,10 @@ func ServeHttp(credentialsManager credentials.Manager, containerInstanceArn stri
 }
 
 // setupServer starts the HTTP server for serving IAM Role Credentials for Tasks.
-func setupServer(credentialsManager credentials.Manager, auditLogger audit.AuditLogger) http.Server {
+func setupServer(credentialsManager credentials.Manager, auditLogger audit.AuditLogger) *http.Server {
 	serverMux := http.NewServeMux()
-	serverMux.HandleFunc(credentials.CredentialsPath, credentialsV1RequestHandler(credentialsManager, auditLogger))
+	serverMux.HandleFunc(credentials.V1CredentialsPath, credentialsV1V2RequestHandler(credentialsManager, auditLogger, getV1CredentialsID, apiVersion1))
+	serverMux.HandleFunc(credentials.V2CredentialsPath+"/", credentialsV1V2RequestHandler(credentialsManager, auditLogger, getV2CredentialsID, apiVersion2))
```

Comment:
A request to `/v2/credentials` would get a `302` to `/v2/credentials/`.  A request to `/v2/credentials/anything/including/slashes` gets handled by `credentialsV1V2RequestHandler`.  See [the `ServeMux` documentation](https://golang.org/pkg/net/http/#ServeMux).